molecularnodes/ui/panel.py

```diff
@@ -162,34 +150,112 @@ def ui_from_node(layout, node):
+    row = layout.row()
+    row = row.split(factor=0.9)
+    row.template_list(
+        "MN_UL_TrajectorySelectionListUI",
+        "A list",
+        bob,
+        "mn_universe_selections",
+        bob.mn,
+        "universe_selection_index",
+        rows=3,
+    )
+    col = row.column()
+    col.operator("mn.universe_selection_add", icon="ADD", text="")
+    col.operator("mda.delete_item", icon="REMOVE", text="")
+    if bob.mn_universe_selections:
+        item = bob.mn_universe_selections[bob.mn.universe_selection_index]
+
+        col = layout.column(align=False)
+        row = col.row()
+        col.prop(item, "selection_str")
+
+        if item.message != "":
+            box = col.box()
+            box.label(text="Invalid Selection", icon="ERROR")
+            box.label(text=item.message)
+            box.alert = True
+            op = box.operator("wm.url_open", text="Selection Langauge Docs", icon="URL")
+            op.url = (
+                "https://docs.mdanalysis.org/stable/documentation_pages/selections.html"
+            )
+
+
 def panel_object(layout, context):
     object = context.active_object
-    mol_type = object.mn.molecule_type
+    try:
+        mol_type = object.mn.molecule_type
+    except AttributeError:
+        return None
     if mol_type == "":
         layout.label(text="No MN object selected")
         return None
-
     if mol_type == "pdb":
         layout.label(text=f"PDB: {object.mn.pdb_code.upper()}")
     if mol_type == "md":
-        layout.prop(object.mn, "subframes")
+        panel_md_properties(layout, context)
     if mol_type == "star":
-        layout.label(text=f"Ensemble")
+        layout.label(text="Ensemble")
         box = layout.box()
         ui_from_node(box, nodes.get_star_node(object))
-        return
-
-    row = layout.row(align=True)
-    row.label(text="Style")
-    current_style = nodes.format_node_name(
-        nodes.get_style_node(object).node_tree.name
-    ).replace("Style ", "")
-    row.operator_menu_enum("mn.style_change", "style", text=current_style)
-    box = layout.box()
-    ui_from_node(box, nodes.get_style_node(object))
+        return None
+
+
+def item_ui(layout, item):
+    row = layout.row()
+    row.label(text=item.name)
+    col = row.column()
+    op = col.operator("mn.session_create_model")
+    op.uuid = item.uuid
+    col.enabled = item.object is None
+
+    op = row.operator("mn.session_remove_item", text="", icon="CANCEL")
+    op.uuid = item.uuid
+
+    if item.object is not None:
+        row = layout.row()
+        row.label(text=f"Object: {item.object.name}", icon="OUTLINER_OB_MESH")
+
+
+def panel_session(layout, context):
+    session = context.scene.MNSession
+    # if session.n_items > 0:
+    #     return None
     row = layout.row()
-    row.label(text="Experimental", icon_value=2)
-    row.operator("mn.add_armature")
+    row.label(text="Loaded items in the session")
+    row.operator("mn.session_reload")
```

Comment:
You are probably aware of this, but this function doesn’t exist.

Reply:
Ah yes - this was from another PR. Will track it down and clean up.